conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinIndexer.scala

```diff
@@ -0,0 +1,146 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import java.util.concurrent.Executors
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+
+import cats.effect.{IO, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import org.http4s.headers.Authorization
+import org.http4s.BasicCredentials
+import org.http4s.client.blaze.BlazeClientBuilder
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+import slickeffect.transactor.{config => transactorConfig}
+
+import tech.cryptonomic.conseil.common.util.DatabaseUtil
+import tech.cryptonomic.conseil.indexer.config.LorreConfiguration
+import tech.cryptonomic.conseil.common.config.Platforms
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinConfiguration
+import tech.cryptonomic.conseil.indexer.LorreIndexer
+import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+
+/**
+  * Class responsible for indexing data for Bitcoin Blockchain.
+  *
+  * @param lorreConf Lorre configuration
+  * @param bitcoinConf Bitcoin configuration
+  */
+class BitcoinIndexer(
+    lorreConf: LorreConfiguration,
+    bitcoinConf: BitcoinConfiguration
+) extends LazyLogging
+    with LorreIndexer
+    with LorreProgressLogging {
+
+  /**
+    * Executor for the rpc client, timer and to handle stop method.
+    */
+  private val indexerExecutor = Executors.newFixedThreadPool(bitcoinConf.batchingConf.indexerThreadsCount)
+
+  /**
+    * Dedicated executor for the http4s.
+    */
+  private val httpExecutor = Executors.newFixedThreadPool(bitcoinConf.batchingConf.httpFetchThreadsCount)
+
+  /**
+    * [[cats.ContextShift]] is the equivalent to [[ExecutionContext]],
+    * it's used by the Cats Effect related methods.
+    */
+  implicit private val contextShift = IO.contextShift(ExecutionContext.fromExecutor(indexerExecutor))
+
+  /**
+    * [[ExecutionContext]] to handle stop method.
+    */
+  implicit private val indexerEC = ExecutionContext.fromExecutor(indexerExecutor)
```

Comment:
A couple of questions

Is this implicitly required by `stop`? maybe we could pass it explicitly, or use the global executor there. I'm thinking if we could remove from the scope and the code an extra value which is needed only once in a specific place...

What happens when the thread running on an executor invokes his own executor's pool shutdown?

Reply:
I've changed code to use implicitly `ExecutionContext.global` in `stop`